### PR TITLE
python311Packages.minari: init at 0.4.3

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Add python package (and application) [minari](https://github.com/Farama-Foundation/Minari), a standard format for offline reinforcement learning datasets, with popular reference datasets and related utilities.

The necessary dependencies are introduced in this same PR.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.gymnasium-robotics</li>
    <li>python312Packages.gymnasium-robotics.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python311Packages.gymnasium-robotics</li>
    <li>python311Packages.gymnasium-robotics.dist</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.gymnasium-robotics</li>
    <li>python312Packages.gymnasium-robotics.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python311Packages.gymnasium-robotics</li>
    <li>python311Packages.gymnasium-robotics.dist</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309833` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>minari (python311Packages.minari)</li>
    <li>minari.dist (python311Packages.minari.dist)</li>
    <li>python311Packages.nbmake</li>
    <li>python311Packages.nbmake.dist</li>
    <li>python311Packages.portion</li>
    <li>python311Packages.portion.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.nbmake</li>
    <li>python312Packages.nbmake.dist</li>
    <li>python312Packages.portion</li>
    <li>python312Packages.portion.dist</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -31399,6 +31399,8 @@ with pkgs;
 
   manuskript = libsForQt5.callPackage ../applications/editors/manuskript { };
 
+  minari = python3Packages.toPythonApplication python3Packages.minari;
```
Comment (drupol) [inline pkgs/top-level/all-packages.nix:31402]: Is there a `meta.mainProgram` automatically set for this?

Comment (GaetanLepage) [reply pkgs/top-level/all-packages.nix:31402]: Not sure but it works.
`manim` is in the same case and does not declare a `mainProgram`.

Comment (GaetanLepage) [reply pkgs/top-level/all-packages.nix:31402]: I have found other python packages doing it so I added it.

Comment (GaetanLepage) [reply pkgs/top-level/all-packages.nix:31402]: > Does running `nix run .#minari` works? If yes, then it's fine.

This was working without and still is.

Comment (drupol) [reply pkgs/top-level/all-packages.nix:31402]: Right, let's leave it like this, it's fine now.

**Pull request merged**